src/pages/api/twitter-request-token.ts

```diff
@@ -0,0 +1,80 @@
+import { createHmac, randomBytes } from "crypto"
+import { NextApiHandler } from "next"
+
+const CONSUMER_KEY = process.env.TWITTER_API_KEY
+const CONSUMER_SECRET = process.env.TWITTER_API_SECRET
+const SIGNATURE_METHOD = "HMAC-SHA1"
+const OAUTH_VERSION = "1.0"
+const DEFAULT_CALLBACK_URL = "https://guild.xyz/oauth"
+const BASE_URL = "https://api.twitter.com/oauth/request_token"
+const KEY = `${encodeURIComponent(CONSUMER_SECRET)}&`
+const RESPONSE_REGEX =
+  /oauth_token=(.*?)&oauth_token_secret=(.*?)&oauth_callback_confirmed=true/
+
+const handler: NextApiHandler = async (req, res) => {
+  if (req.method?.toLowerCase() !== "get") {
```

Comment:
Really minor thing, but the we could use the `Allow` header here like we do on our other serverless endpoints ;)

```ts
if (req.method !== "GET") {
    res.setHeader("Allow", "GET")
    res.status(405).json({ error: `Method ${req.method} is not allowed` })
    return
}
```

Reply:
Wow, thank you! What an elegant way to handle this!
Made this change in https://github.com/agoraxyz/guild.xyz/pull/852/commits/616e4b3e51182f6ee2d2bf6f6678a42dc84b7711